packages/nest-commander/src/command-runner.interface.ts

```diff
@@ -23,7 +24,20 @@ export type InquirerKeysWithPossibleFunctionTypes =
 
 type InquirerQuestionWithoutFilter<T> = Omit<T, 'filter'>;
 
+type CommandRunnerClass = ClassProvider<CommandRunner> & typeof CommandRunner;
+
 export abstract class CommandRunner {
+  static registerCommands(meta: string = CommandMeta): CommandRunnerClass[] {
```

Comment:
Could we call this `registerWithSubCommands`? That way the intent is clear. The code is actually really clean, and if the command is clear enough then I don't see why we can't add it in. Thanks for adding the tests too, showing it works is really nice.

The only other thing I would ask is can you update the documentation? Just a new section in the [`Commander`](https://nest-commander.jaymcdoniel.dev/en/features/commander/) section would do